foundations/src/telemetry/log/init.rs

```diff
@@ -95,12 +96,29 @@ pub(crate) fn init(service_info: &ServiceInfo, settings: &LoggingSettings) -> Bo
         }
         (LogOutput::File(file), LogFormat::Json) => {
             let buf = BufWriter::with_capacity(BUF_SIZE, File::create(file)?);
-            let drain = build_json_log_drain(buf);
+            let drain = build_json_log_drain(buf, false);
             AsyncDrain::new(drain).chan_size(CHANNEL_SIZE).build()
         }
+        (LogOutput::Stdout, log_fmt) => {
+            let stdout = unsafe { File::from_raw_fd(STDOUT_FILENO) };
+            let buf = BufWriter::with_capacity(BUF_SIZE, stdout);
+
+            match log_fmt {
+                LogFormat::Text => {
+                    let drain = TextDrain::new(PlainDecorator::new(buf))
+                        .build()
+                        .fuse();
+                    AsyncDrain::new(drain).chan_size(CHANNEL_SIZE).build()
+                },
+                LogFormat::Json => {
+                    let drain = build_json_log_drain(buf, false);
```

Comment:
We need to keep `flush` option enabled on JSON drain. It guarantees that JSON record will be flushed once finished even if underlying BufWriter is not full: https://docs.rs/slog-json/latest/slog_json/struct.JsonBuilder.html#method.set_flush